jenkins-build-pack-website/Jenkinsfile

```diff
@@ -628,10 +628,11 @@ def updateCloudFrontPolicy(String service, String s3Bucket){
 
 	def bucketPolicy = loadBucketPolicy(s3Bucket)
 	def statements = bucketPolicy.Statement
+	def CFID = getCloudFrontDetails(environment, service)
 
-//	def principal = ["AWS": env.CLOUDFRONT_OAI]
+	//	def principal = ["AWS": env.CLOUDFRONT_OAI]
 	//def principal = ["AWS": "*"]
-	def principal = "*"
+	def principal = {"AWS": "arn:aws:iam::cloudfront:user/CloudFront Origin Access Identity ${CFID[0]}"}
```

Comment:
Validate that this is CF origin access identity and not Distribution Id

Reply:
@deepusundar  is correct. This needs to be CF origin access identity. you can get it from aws cli:

https://docs.aws.amazon.com/cli/latest/reference/cloudfront/list-cloud-front-origin-access-identities.html

where you will need to look up using comment field. comment field has format {env-prefix}-origin_access_identity

Reply:
def outputStr = sh (
                        script: "aws cloudfront list-cloud-front-origin-access-identities --output text \
									--query \"CloudFrontOriginAccessIdentityList.Items[?Comment=='${configLoader.env_name_prefix}-origin_access_identity'].{j: Id}\"",
                        returnStdout: true
                    ).trim()

                echo outputStr

Reply:
You can see the values in the console here: https://console.aws.amazon.com/cloudfront/home?region=us-east-1#oai: